### PR TITLE
Feature: [Review Sort] JS/Servlet/HTML Cohesion

### PR DESCRIPTION
**Pull request by dterr**
### Ability to Sort Comments by Recency or Rating
This uses the previously created buttons in the Reviews modal to communicate with the backend.
_Button -> Sort Function -> ReFetch with Sort -> Query and Organize -> Present on front end._
This takes advantage of using multiple parameters in the HTTPServletRequest. The `List<Comment>` is sorted within `PlaceReviews` for front-end presentation. This passes basic testing, but not robust.

Changes:

- `Comment.java` Created comparator for ordering by metascore.
- `PlaceReviews.java` Introduce public sorting method `sortReviews(String sortType)`.
- `ReviewServlet.java` Add sortType field to HTTPServletRequest and utilize it.
- `index-script.js` Create comment sorting functions that are wired to appropriate buttons.
- `index.html` Start button group off as hidden.
- `modal-overlay.css` More subtle styling for active button.

### CODE REVIEW

File: src/main/webapp/index-script.js [hunk 1]
```diff
@@ -45,11 +46,15 @@ function loadMainButtons() {
...
-    // TODO: Make sort relevant function
+    commentSortRelevant.classList.add("active");
+    commentSortRecent.classList.remove("active");
+    resortReviews(prev_ID, 'relevant');
   });
   
   commentSortRecent.addEventListener("click", () => {
-    // TODO: Make sort recent function
+    commentSortRecent.classList.add("active");
+    commentSortRelevant.classList.remove("active");
```
Comment (kcirerick) [inline src/main/webapp/index-script.js:56]: Does this mean you're adding the button after they click on it? Shouldn't you be hiding the button that gets clicked on and revealing the opposite option instead?

File: src/main/webapp/index-script.js [hunk 2]
```diff
@@ -393,8 +404,9 @@ function displayReviewModal() {
  * Queries ReviewServlet with elementID// Comment object gets it's datastore id. 
     comment.setId(reviewEntity.getKey().getId()); to find internal datastore
  */
-function fetchReviews(placeID) {
-  const request = '/review?place_id=' + placeID;
+function fetchReviews(placeID, sort='recent') {
+  prev_ID = placeID;
```
Comment (ccjeane) [inline src/main/webapp/index-script.js:408]: WARNING: Possible Merge Conflict, no biggie though.

File: src/main/java/com/google/sps/servlets/ReviewServlet.java [hunk 3]
```diff
@@ -34,6 +34,8 @@
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the requested place using it's ID. 
     String place_id = request.getParameter("place_id");
+    // Defer to sort by recent if poor format
```
Comment (ccjeane) [inline src/main/java/com/google/sps/servlets/ReviewServlet.java:37]: Nice!

File: src/main/webapp/index-script.js (hunk 1 above)
Comment (dterr) [reply src/main/webapp/index-script.js:56]: No this means that it adds the active styling for the button on click.
Activate styling just means that the button is now colored in a bright color to distinguish itself from the other.

**Comment by ccjeane** [comment]
Rapid-fire-clicking, didn't mean to delete and restore

**Pull request merged**